### PR TITLE
chore(CI): Wait on webhookserver deployment before using pod

### PR DESCRIPTION
**Pull request by mtodor**
### Description

We had a CI failure with the following logs:
```
Release "webhookserver" does not exist. Installing it now.
NAME: webhookserver
LAST DEPLOYED: Mon Dec 16 02:10:36 2024
NAMESPACE: stackrox
STATUS: deployed
REVISION: 1
TEST SUITE: None
/go/src/github.com/stackrox/stackrox
Got pod 
****
**** 02:10:42: ERROR: test failed [Test failed: exit 1] [QA tests part I]
```

This indicates that pod didn't exist at the point where we looked for it in this code section:
```
    pod="$(kubectl -n stackrox get pod -l app=webhookserver -o name)"
    echo "Got pod ${pod}"
    [[ -n "${pod}" ]]
```

We are not checking if the deployment has any pods. This PR adds a wait for the deployment to be ready before continuing to use a pod for port forward. 

### User-facing documentation

- [x] CHANGELOG is updated **OR** update is not needed
- [x] [documentation PR](https://spaces.redhat.com/display/StackRox/Submitting+a+User+Documentation+Pull+Request) is created and is linked above **OR** is not needed

### Testing and quality

- [x] the change is production ready: the change is GA or otherwise the functionality is gated by a feature flag
- [ ] CI results are inspected

#### Automated testing

- [x] no test changes

#### How I validated my change

Tested helm and command on a GKE cluster.

### CODE REVIEW

**Review by rhybrillou — APPROVED** [review]

File: scripts/ci/create-webhookserver.sh [hunk 1]
```diff
@@ -30,7 +30,8 @@ install_webhook_server() {
 }
 
 create_webhook_server_port_forward() {
-    sleep 5
+    # Ensure webhookserver state is stable before continuing with port forward.
+    kubectl rollout status deployment webhookserver --namespace stackrox --timeout=5m --watch=true
     pod="$(kubectl -n stackrox get pod -l app=webhookserver -o name)"
```
Comment (porridge) [inline scripts/ci/create-webhookserver.sh:35]: Maybe while at it:
```suggestion
    local pod
    pod="$(kubectl -n stackrox get pod -l app=webhookserver -o name)"
```

Comment (porridge) [reply scripts/ci/create-webhookserver.sh:35]: Perhaps also similar with `log=` below (GH won't let me comment there ☹️ ).

**Review by porridge — APPROVED** [review]
I wonder whether we need `kubectl -n stackrox wait --for=condition=ready "${pod}"` any more, but probably does not hurt to leave it around.

**Pull request merged**